torch/testing/_internal/common_modules.py

```diff
@@ -678,6 +678,29 @@ def module_inputs_torch_nn_TransformerEncoderLayer(module_info, device, dtype, r
                 desc='no_batch_dim'
             ))
 
+    # test fast path. We abuse the fact that we can supply an arbitrary reference_fn to
+    # force the fast path to run -- we cannot call .eval() through the ordinary
+    # ModuleInfo interface, so the "reference" is actually the code under test.
```

Comment:
We should just fix the `ModuleInfo` interface so it's flexible enough to do what you need here.

Maybe an enum on `ModuleInput` specifying whether it should be used for training / evaluation / both (default) ? If this works okay, I'll open a PR for the support.

Reply:
Addressing this in #78735.

Reply:
@swolchok FYI this landed. Tests now run in both train and eval, and you can add eval-only sample inputs